src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -35,32 +36,34 @@ class WidgetController(
     )
 
     @PostMapping("/$WIDGET_START_IDENT_BTN_CLICKED")
-    fun handleStartIdentButtonClicked(@RequestParam(required = false, name = "hash") sessionHash: String?): ResponseEntity<String> {
+    fun handleStartIdentButtonClicked(@RequestParam(required = false, name = "hash") sessionHash: String?, @RequestHeader(HttpHeaders.USER_AGENT) userAgent: String): ResponseEntity<String> {
```

Comment:
A `@RequestHeader` is required by default (if you do not explicitly set `required=false`. Either make this parameter optional, or take it for granted that it is never null since you use `String?` when handing the parameter to the `publishMatomoEvent` method. 
Since it is completely up to the caller (i.e. user) if the `user-agent` is set, I would suggest to make the parameter optional since otherwise the caller will get a `400 Bad Request` response.

Reply:
Good catch! I marked it as optional by using String?. 
Tested it and routes work when not sending any user agent.